src/m365/entra/commands/group/group-set.ts

```diff
@@ -0,0 +1,330 @@
+      }
+
+      const requestBody = this.createRequestBody(args.options);
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}/v1.0/groups/${groupId}`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        data: requestBody
+      };
+
+      await request.patch(requestOptions);
+
+      ownerIds = await this.getUserIds(logger, args.options.ownerIds, args.options.ownerUserNames);
+      if (ownerIds.length !== 0) {
+        await this.updateUsers(groupId!, 'owners', ownerIds);
+      }
+      else if (this.verbose) {
+        await logger.logToStderr(`No owners to update.`);
+      }
+
+      memberIds = await this.getUserIds(logger, args.options.memberIds, args.options.memberUserNames);
+      if (memberIds.length !== 0) {
+        await this.updateUsers(groupId!, 'members', memberIds);
+      }
+      else if (this.verbose) {
+        await logger.logToStderr(`No members to update.`);
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  };
+
+  private createRequestBody(options: Options): any {
+    const requestBody: any = {
+      displayName: options.newDisplayName,
+      description: options.description === '' ? null : options.description,
+      mailNickName: options.mailNickname,
+      visibility: options.visibility
+    };
+
+    this.addUnknownOptionsToPayload(requestBody, options);
+    return requestBody;
+  }
+
+  private async getUserIds(logger: Logger, userIds?: string, userNames?: string): Promise<string[]> {
+    if (userIds) {
+      return formatting.splitAndTrim(userIds);
+    }
+
+    if (userNames) {
+      if (this.verbose) {
+        await logger.logToStderr(`Retrieving users ids...`);
+      }
+      return entraUser.getUserIdsByUpns(formatting.splitAndTrim(userNames));
+    }
+
+    return [];
+  }
+
+  private async updateUsers(groupId: string, role: 'members' | 'owners', userIds: string[]): Promise<void> {
+    const groupUsers = await odata.getAllItems<User>(`${this.resource}/v1.0/groups/${groupId}/${role}/microsoft.graph.user?$select=id`);
+    const userIdsToAdd = userIds.filter(userId => !groupUsers.some(groupUser => groupUser.id === userId));
+    const userIdsToRemove = groupUsers.filter(groupUser => !userIds.some(userId => groupUser.id === userId));
+
+    for (let i = 0; i < userIdsToAdd.length; i += 400) {
+      const userIdsBatch = userIdsToAdd.slice(i, i + 400);
+      const requestOptions = this.getRequestOptions();
+
+      // only 20 requests per one batch are allowed
+      for (let j = 0; j < userIdsBatch.length; j += 20) {
+        // only 20 users can be added in one request
+        const userIdsChunk = userIdsBatch.slice(j, j + 20);
+        requestOptions.data.requests.push({
+          id: j + 1,
+          method: 'PATCH',
+          url: `/groups/${groupId}`,
+          headers: {
+            'content-type': 'application/json;odata.metadata=none'
```

Comment:
Let's also add an accept header.

Reply:
Maybe group add and group set commands should be consistent in this. I've added the accept header.